gui/src/app/pages/HomePage/SaveProjectWindow.tsx

```diff
@@ -0,0 +1,208 @@
+            </tr>
+          ))}
+        </tbody>
+      </table>
+      <div>&nbsp;</div>
+      {!exportingToGist && (
+        <div>
+          <button
+            onClick={async () => {
+              serializeAsZip(data).then(([zipBlob, name]) =>
+                triggerDownload(zipBlob, `SP-${name}.zip`, onClose),
+              );
+            }}
+          >
+            Save to .zip file
+          </button>
+          &nbsp;
+          <button
+            onClick={() => {
+              setExportingToGist(true);
+            }}
+          >
+            Save to GitHub Gist
+          </button>
+        </div>
+      )}
+      {exportingToGist && (
+        <GistExportView
+          fileManifest={fileManifest}
+          title={data.meta.title}
+          onClose={onClose}
+        />
+      )}
+    </div>
+  );
+};
+
+type EditTitleComponentProps = {
+  value: string;
+  onChange: (value: string) => void;
+};
+
+const EditTitleComponent: FunctionComponent<EditTitleComponentProps> = ({
+  value,
+  onChange,
+}) => {
+  return (
+    <input
+      type="text"
+      value={value}
+      onChange={(e) => onChange(e.target.value)}
+    />
+  );
+};
+
+type GistExportViewProps = {
+  fileManifest: Partial<FileRegistry>;
+  title: string;
+  onClose: () => void;
+};
+
+const GistExportView: FunctionComponent<GistExportViewProps> = ({
+  fileManifest,
+  title,
+  onClose,
+}) => {
+  const [gitHubPersonalAccessToken, setGitHubPersonalAccessToken] =
+    useState("");
+  const [gistUrl, setGistUrl] = useState<string | null>(null);
+
+  const handleExport = useCallback(async () => {
+    try {
+      const gistUrl = await saveAsGitHubGist(fileManifest, {
+        defaultDescription: title,
+        personalAccessToken: gitHubPersonalAccessToken,
+      });
+      setGistUrl(gistUrl);
+    } catch (err: any) {
+      alert(`Error saving to GitHub Gist: ${err.message}`);
+    }
+  }, [gitHubPersonalAccessToken, fileManifest, title]);
+
+  return (
+    <div style={{ maxWidth: 800 }}>
+      <h3>Save to GitHub Gist</h3>
+      <p>
+        In order to save this project as a GitHub Gist, you will need to provide
+        a GitHub Personal Access Token.&nbsp; This token will be used to
+        authenticate with GitHub and create a new Gist with the files in this
+        project.&nbsp; You can create a new Personal Access Token by visiting
+        your{" "}
+        <a href="https://github.com/settings" target="_blank" rel="noreferrer">
+          GitHub settings
+        </a>
+        .&nbsp; Go to <i>Developer settings</i> and <i>Tokens (classic)</i>
+        .&nbsp; Generate a new classic token and be sure to only grant gist
+        scope with an expiration date.&nbsp; Copy the token and paste it into
+        the field below.
+      </p>
+      <p>
+        For security reasons, your token will not be saved in this
+        application,&nbsp; so you may want to store it securely in a text file
+        for future use.
```

Comment:
I thought we had decided local storage was probably fine for this?

Reply:
I think it is okay, but if it's okay, let's make that a future feature. I want to think through security implications more carefully and how to present this to the user. I think saving in browser should be opt-in only.